File: src/main/java/com/gjw/go/common/exception/AppException.java
package com.gjw.go.common.exception;

import com.gjw.go.common.enums.SysCodeEnums;
import lombok.Getter;

/**
 * 自定义异常
 */
@Getter
public class AppException extends RuntimeException {
    /**
     * 错误码
     */
    private String code;
    /**
     * 错误描述
     */
    private String msg;

    public static AppException error() {
        return new AppException(SysCodeEnums.ERROR.getCode(), SysCodeEnums.ERROR.getMsg());
    }

    public static AppException error(String msg) {
        return new AppException(SysCodeEnums.ERROR.getCode(), msg);
    }

    public static AppException success() {
        return new AppException(SysCodeEnums.SUCCESS.getCode(), SysCodeEnums.SUCCESS.getMsg());
    }

    public static AppException success(String msg) {
        return new AppException(SysCodeEnums.SUCCESS.getCode(), msg);
    }

    public AppException(String code, String msg) {
        super(String.format("%s:%s", code, msg));
        this.code = code;
        this.msg = msg;
    }
}

File: src/main/kotlin/com/gjw/go/common/exception/AppException.kt
package com.gjw.go.common.exception

import com.gjw.go.common.enums.SysCodeEnums

class AppException : RuntimeException() {
    var code: String? = null
    var msg: String? = null

    companion object {
        fun error(msg:String?):AppException{
            return AppException().apply {
                this.code = SysCodeEnums.ERROR.code
                this.msg = msg?:SysCodeEnums.ERROR.msg
            }
        }
        fun assertNotNull(any: Any,msg: String){
            if(any == null){
                throw error(msg)
            }
        }
    }

    override fun toString(): String {
        return "AppException(code=$code, msg=$msg)"
    }
}